[CI] GenerativeIT test failing

**Build Scans:**
- [elasticsearch-periodic #9440 / encryption-at-rest](https://gradle-enterprise.elastic.co/s/wkdca7x5wlpig)
- [elasticsearch-pull-request #81658 / part-3](https://gradle-enterprise.elastic.co/s/6f2ihajsjr2ty)
- [elasticsearch-pull-request #81618 / part-3](https://gradle-enterprise.elastic.co/s/zera3dkqfhl4y)
- [elasticsearch-pull-request #81599 / part-3](https://gradle-enterprise.elastic.co/s/zd7lmk2svy2uw)
- [elasticsearch-pull-request #81412 / part-3](https://gradle-enterprise.elastic.co/s/s4lh35jksrj2a)
- [elasticsearch-pull-request #81412 / part-3](https://gradle-enterprise.elastic.co/s/b3jcwiwdj2ao2)
- [elasticsearch-pull-request #81203 / part-3](https://gradle-enterprise.elastic.co/s/54stfw3t7rjza)
- [elasticsearch-pull-request #81148 / part-3](https://gradle-enterprise.elastic.co/s/homfs4jfrqc5w)
- [elasticsearch-pull-request #81148 / part-3](https://gradle-enterprise.elastic.co/s/5ervug4elxr3s)
- [elasticsearch-pull-request #80976 / part-3](https://gradle-enterprise.elastic.co/s/lnliyfzosxw5m)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT.test" -Dtests.seed=B184F0960CE6BD0D -Dtests.locale=ko-KP -Dtests.timezone=Etc/GMT-13 -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.multi_node.GenerativeIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: query: from mv_sample_data | dissect message "%{qpRbPZwc} %{UbWGGBtGlG} %{DVLbsKYvS}" | mv_expand `qpRbPZwc` | CHANGE_POINT event_duration ON @timestamp AS `qpRbPZwc`, `UbWGGBtGlG` | keep qpRbPZwc, @timestamp | where false AND false AND false| rename qpRbPZwc as message | lookup join message_types_lookup on message| rename type as message | lookup join message_types_lookup on message | sort type DESC | limit 7896 | rename `@timestamp` AS `VGjcBqSUqiMg` | rename message AS ZvuHgaUFfZff, `ZvuHgaUFfZff` AS type | keep VGjcBqSUqiMg, `type`| rename type as message | lookup join message_types_lookup on message
exception: method [POST], host [http://[::1]:46381], URI [/_query?format=yaml&pretty=true&error_trace=true], status line [HTTP/1.1 500 Internal Server Error]
---
error:
  root_cause:
  - type: "illegal_state_exception"
    reason: "Found 1 problem\nline 1:562: Plan [Join[LEFT,[message{r}#2951],[message{r}#2951],[message{f}#2960]]]\
      \ optimized incorrectl
[truncated]
```

**Issue Reasons:**
- [main] 11 failures in test test (1.7% fail rate in 658 executions)
- [main] 9 failures in step part-3 (2.9% fail rate in 307 executions)
- [main] 7 failures in pipeline elasticsearch-pull-request (2.3% fail rate in 303 executions)
- [main] 2 failures in pipeline elasticsearch-periodic (20.0% fail rate in 10 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -469,9 +469,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.action.EsqlActionBreakerIT
   method: testFromEvalStats
   issue: https://github.com/elastic/elasticsearch/issues/131503
-- class: org.elasticsearch.xpack.esql.qa.single_node.GenerativeIT
-  method: test
-  issue: https://github.com/elastic/elasticsearch/issues/131508
 - class: org.elasticsearch.xpack.downsample.DownsampleWithBasicRestIT
   method: test {p0=downsample-with-security/10_basic/Downsample index}
   issue: https://github.com/elastic/elasticsearch/issues/131513

--- a/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
+++ b/x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/generative/GenerativeRestTest.java
@@ -50,6 +50,7 @@ public abstract class GenerativeRestTest extends ESRestTestCase {
         "Plan \\[ProjectExec\\[\\[<no-fields>.* optimized incorrectly due to missing references", // https://github.com/elastic/elasticsearch/issues/125866
         "The incoming YAML document exceeds the limit:", // still to investigate, but it seems to be specific to the test framework
         "Data too large", // Circuit breaker exceptions eg. https://github.com/elastic/elasticsearch/issues/130072
+        "optimized incorrectly due to missing references", // https://github.com/elastic/elasticsearch/issues/131509
 
         // Awaiting fixes for correctness
         "Expecting at most \\[.*\\] columns, got \\[.*\\]" // https://github.com/elastic/elasticsearch/issues/129561

